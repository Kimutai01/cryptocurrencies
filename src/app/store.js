import { configureStore } from '@reduxjs/toolkit';
import cryptoReducer from '../feature/cryptoSlice';

const store = configureStore({
  reducer: {
    crypto: cryptoReducer,
  },
});

export default store;
