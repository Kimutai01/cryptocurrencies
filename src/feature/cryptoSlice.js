import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import axios from 'axios';

export const getCrypto = createAsyncThunk('crypto/getCrypto', async () => {
  const getResponse = await axios.get(
    'https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=50&page=1&sparkline=false',
  );
  const dat = getResponse.data;
  return dat;
});

const cryptoSlice = createSlice({
  name: 'crypto',
  initialState: [],
  reducers: {},
  extraReducers: {
    [getCrypto.fulfilled]: (state, action) => action.payload,
  },
});

export default cryptoSlice.reducer;
