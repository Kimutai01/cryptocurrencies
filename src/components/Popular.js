import { useEffect } from "react";
import { useDispatch } from "react-redux";
import { useSelector } from "react-redux";
import { Link } from "react-router-dom";
import { getCrypto } from "../feature/cryptoSlice";
import Nav from "../pages/Navbar";
import "./popular.css";

const Popular = () => {
  const dispatch = useDispatch();
  const allPopular = useSelector((state) => state.crypto);
  console.log(allPopular);
  useEffect(() => {
    dispatch(getCrypto());
  }, [dispatch]);
  return (
    <div>
      <Nav />
      <div className="popular-div">
        {allPopular.map((popular, i) => (
          <Link to={`/popularDetail/${popular.id}`} key={i} className="pop">
            <h3 className="name">{popular.name}</h3>
            <img src={popular.image} alt="crypto" />
            <div className="details">
              <p className="change">{popular.symbol}</p>
              <p className="price">{popular.current_price}$</p>
            </div>
          </Link>
        ))}
      </div>
    </div>
  );
};

export default Popular;
