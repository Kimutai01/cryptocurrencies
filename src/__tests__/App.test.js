import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import store from '../app/store';
import Popular from '../components/Popular';
import { getCrypto } from '../redux/cryptoSlice';

test('Popular component testing', () => {
  const TREE = TestRenderer.create(
    <Provider store={store}>
      <Popular />
    </Provider>,
  );
  expect(TREE).toMatchSnapshot();
});

it('getCrypto should return a promise', () => {
  expect(getCrypto()).toEqual(expect.any(Function));
});
