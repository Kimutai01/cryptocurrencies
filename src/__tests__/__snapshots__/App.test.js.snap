// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popular component testing 1`] = `
<div>
  <div>
    <div
      className="carousel slide carousel-fade"
      onKeyDown={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="carousel-indicators"
      >
        <button
          aria-current={true}
          aria-label="Slide 1"
          className="active"
          data-bs-target=""
          onClick={[Function]}
          type="button"
        />
        <button
          aria-current={false}
          aria-label="Slide 2"
          data-bs-target=""
          onClick={[Function]}
          type="button"
        />
        <button
          aria-current={false}
          aria-label="Slide 3"
          data-bs-target=""
          onClick={[Function]}
          type="button"
        />
      </div>
      <div
        className="carousel-inner"
      >
        <div
          className="active carousel-item"
        >
          <img
            alt="stock"
            className="d-block w-100"
            src="tether.jpg"
            style={
              Object {
                "height": "100vh",
              }
            }
          />
          <div
            className="carousel-caption"
          >
            <h3>
              Check popular stocks of varios companies ðŸ’¯
            </h3>
            <p>
              We provide accurate stocks of various companies all over the world.
            </p>
          </div>
        </div>
        <div
          className="carousel-item"
        >
          <img
            alt="stock2"
            className="d-block w-100"
            src="main.jpg"
            style={
              Object {
                "height": "100vh",
              }
            }
          />
          <div
            className="carousel-caption"
          >
            <h3>
              Check Gainers and losers ðŸ˜‡
            </h3>
            <p>
              Avoid losses by checking out the gainers and losers of the day.
            </p>
          </div>
        </div>
        <div
          className="carousel-item"
        >
          <img
            alt="stock4"
            className="d-block w-100"
            src="coin.jpg"
            style={
              Object {
                "height": "100vh",
              }
            }
          />
          <div
            className="carousel-caption"
          >
            <h3>
              Get accurate data of all stocksðŸ‘Œ
            </h3>
            <p>
              We provide all stocks data
            </p>
          </div>
        </div>
      </div>
      <a
        className="carousel-control-prev"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          aria-hidden="true"
          className="carousel-control-prev-icon"
        />
        <span
          className="visually-hidden"
        >
          Previous
        </span>
      </a>
      <a
        className="carousel-control-next"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          aria-hidden="true"
          className="carousel-control-next-icon"
        />
        <span
          className="visually-hidden"
        >
          Next
        </span>
      </a>
    </div>
  </div>
  <div
    className="popular-div"
  />
</div>
`;
